util/rate_limiter.cc

```diff
@@ -136,24 +138,24 @@ GenericRateLimiter::~GenericRateLimiter() {
 }
 
 Status GenericRateLimiter::PrepareOptions(const ConfigOptions& options) {
+  MutexLock g(&request_mutex_);
   if (options_.fairness <= 0) {
     return Status::InvalidArgument("Fairness must be > 0");
   } else if (options_.max_bytes_per_sec <= 0) {
     return Status::InvalidArgument("max_bytes_per_sec must be > 0");
   } else if (options_.refill_period_us <= 0) {
     return Status::InvalidArgument("Refill_period_us must be > 0");
   }
-  Initialize();
+  InitializeLocked();
   return RateLimiter::PrepareOptions(options);
 }
 
 // This API allows user to dynamically change rate limiter's bytes per second.
 void GenericRateLimiter::SetBytesPerSecond(int64_t bytes_per_second) {
   assert(bytes_per_second > 0);
-  rate_bytes_per_sec_ = bytes_per_second;
-  refill_bytes_per_period_.store(
-      CalculateRefillBytesPerPeriod(bytes_per_second),
-      std::memory_order_relaxed);
+  ConfigureFromMap(ConfigOptions(),
+                   {{"rate_bytes_per_sec", std::to_string(bytes_per_second)}})
```

Comment:
This doesn't work in LITE mode and is breaking the LITE test

Reply:
Fixed. I did not find a thread-safe way to update the options of Configurable objects, however, it does not seem to be a new issue and is at least documented now.